applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -177,12 +189,18 @@ static struct bt_audio_stream_ops stream_ops = { .started = stream_started_cb,
 static bool scan_recv_cb(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad,
 			 uint32_t broadcast_id)
 {
-	static char name[DEVICE_NAME_PEER_LEN];
+	char name[MAX(sizeof(CONFIG_BT_DEVICE_NAME), sizeof(CONFIG_BT_DEVICE_NAME_ALT))] = { '\0' };
+	struct bt_name bis_name = { &name[0], ARRAY_SIZE(name) };
 
-	bt_data_parse(ad, adv_data_parse, (void *)name);
-	if (strncmp(name, CONFIG_BT_DEVICE_NAME, DEVICE_NAME_PEER_LEN) == 0) {
-		LOG_INF("Broadcast source %s found", name);
-		return true;
+	bt_data_parse(ad, adv_data_parse, (void *)&bis_name);
+
+	if (strlen(bis_name.name) ==
+	    strlen(brdcast_src_names[active_stream.brdcast_src_name_idx])) {
+		if (strncmp(bis_name.name, brdcast_src_names[active_stream.brdcast_src_name_idx],
```

Comment:
We might need a timeout mechanism for avoiding stuck in a situation when there's no second BIS gateway.